static/js/recent_senders.js

```diff
@@ -15,25 +18,42 @@ exports.process_message_for_senders = function (message) {
     }
 
     topic_dict.set(message.subject, sender_timestamps);
-    senders.set(stream_id, topic_dict);
+    topic_senders.set(stream_id, topic_dict);
+
+    // Process most recent sender to whole stream
+    sender_timestamps = stream_senders.get(stream_id) || new Dict();
+    old_timestamp = sender_timestamps.get(message.sender_id);
+
+    if (old_timestamp === undefined || old_timestamp < message.timestamp) {
+        sender_timestamps.set(message.sender_id, message.timestamp);
+    }
+
+    stream_senders.set(stream_id, sender_timestamps);
 };
 
 exports.compare_by_recency = function (user_a, user_b, stream_id, topic) {
     stream_id = stream_id.toString();
 
-    var topic_dict = senders.get(stream_id);
+    var a_timestamp;
+    var b_timestamp;
+
+    var topic_dict = topic_senders.get(stream_id);
     if (topic !== undefined && topic_dict !== undefined) {
         var sender_timestamps = topic_dict.get(topic);
         if (sender_timestamps !== undefined) {
-            var b_timestamp = sender_timestamps.get(user_b.user_id) || Number.NEGATIVE_INFINITY;
-            var a_timestamp = sender_timestamps.get(user_a.user_id) || Number.NEGATIVE_INFINITY;
+            b_timestamp = sender_timestamps.get(user_b.user_id) || Number.NEGATIVE_INFINITY;
+            a_timestamp = sender_timestamps.get(user_a.user_id) || Number.NEGATIVE_INFINITY;
             return b_timestamp - a_timestamp;
```

Comment:
I think this return statement will still fire if anyone has sent to this topic, even though neither of the users in question have, right?  I think we probably want this to have an `if a.timestamp !== b.timestamp` around the return here...